app/gui/src/controller/ide/desktop.rs

```diff
@@ -48,44 +48,49 @@ pub struct Handle {
 }
 
 impl Handle {
-    /// Create a project controller handle with already loaded project model.
-    pub fn new_with_project(
+    /// Create IDE controller. If `maybe_project_name` is `Some`, a project with provided name will
+    /// be opened. Otherwise controller will be used for project manager operations by Welcome
+    /// Screen.
+    pub async fn new(
         project_manager: Rc<dyn project_manager::API>,
-        initial_project: model::Project,
```

Comment:
I don't like removing the function taking initial project model; it could be handy in tests to provide already mocked model::Project without setting up mock of project_manager. Therefore, please:
1. keep the old `new_with_project` method
2. The `new` constructor should use new_with_project (as `new_with_opened_project did before).

Reply:
I added `new_with_project_model` constructor (though it accepts `Option<model::Project>` or code reuse with `new` won't be possible).